network.go

```diff
@@ -622,33 +632,41 @@ func (n *network) Delete() error {
 		return &UnknownNetworkError{name: name, id: id}
 	}
 
-	numEps := n.getEpCnt().EndpointCnt()
-	if numEps != 0 {
+	if !force && n.getEpCnt().EndpointCnt() != 0 {
 		return &ActiveEndpointsError{name: n.name, id: n.id}
 	}
 
-	if err = n.deleteNetwork(); err != nil {
-		return err
+	// Mark the network for deletion
+	if !n.inDelete && n.Type() != "host" && n.Type() != "null" {
```

Comment:
@aboch this isnt anything new that we introduced in Docker. Its been done at Docker level from 1.9. Hence am curious on why you added this now ?